server/src/authority/authority.rs

```diff
@@ -1028,7 +1053,19 @@ impl Authority {
             }
         };
 
-        result
+        // This is annoying. The 1035 spec literally specifies that most DNS authorities would want to store
+        //   records in a list except when there are a lot of records. But this makes indexed lookups by name+type
+        //   always return empty sets. This is only important in the negative case, where other DNS authorities
+        //   generally return NoError and no results when other types exist at the same name. bah.
+        if result.is_empty() {
```

Comment:
This is where, IMO, the stupidity of the spec is detected, for proper response handling later...